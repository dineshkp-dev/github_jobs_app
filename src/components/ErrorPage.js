import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/error.css';

function ErrorPage(props) {
  const errorMessage =
    (props && props.errorMessage) || 'Oops! An error occured.';
  return (
    <div className='error-container'>
      <h1 className='error-title'>Error!</h1>
      <div className='error-msg'>
        <div className='error-text'>{errorMessage}</div>
        Please click <Link to='/'>here</Link> to go back to the home page
      </div>
    </div>
  );
}

export default ErrorPage;
