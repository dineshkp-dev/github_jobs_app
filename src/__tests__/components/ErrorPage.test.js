import ErrorPage from '../../components/ErrorPage';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('ErrorBoundary snapshot test', () => {
  test('renders', () => {
    const { container } = render(
      <Router>
        <ErrorPage />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
