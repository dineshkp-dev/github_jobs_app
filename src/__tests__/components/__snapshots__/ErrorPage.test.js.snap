// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary snapshot test renders 1`] = `
<div>
  <div
    class="error-container"
  >
    <h1
      class="error-title"
    >
      Error!
    </h1>
    <div
      class="error-msg"
    >
      <div
        class="error-text"
      >
        Oops! An error occured.
      </div>
      Please click 
      <a
        href="/"
      >
        here
      </a>
       to go back to the home page
    </div>
  </div>
</div>
`;
